src/test/java/com/opentable/versionedconfig/GitServiceTest.java

```diff
@@ -182,6 +190,10 @@ public void testFallbackPull() throws Exception {
             assertThat(vcu.getNewRevisionMetadata()).isNotNull();
             assertThat(changeNames(vcu)).contains("foo.txt");
         }
+        assertThat(ca.captured).anyMatch(e ->
+                e.getLevel() == Level.WARN &&
+                e.getFormattedMessage().contains("While fetching remote git://example.invalid")
+        );
```

Comment:
Nice! Seems crazy if querying logs is the easiest way to do this, but this reads well nonetheless... I blame "the previous guy" :P

Reply:
We could redesign the interface to make it more testable, expose some kind of "tried these remotes".  But it's a lot of design when this works just fine.